### DISCUSSION TITLE
아이템 먹기

### DISCUSSION THREAD

**Discussion by almondsky** [naverz/zepeto-studio-kor#846] [4. General]
안녕하세요, 

코인먹기 월드를 만들어 보려 하는데, 제페토 샘플북에서 아이템 먹기 스크립트를 가져와 이렇게 스크립트를 복사 붙여넣기를 했구, 코인먹기에 사용될 코인은 현재 프리팹으로 만들어 놓구 스크립트를 넣어준 상태입니다. 

아이템 먹기 스크립트를 가져와 이렇게 붙여넣었을 때, 
<img width="921" alt="Screen Shot 2022-06-06 at 7 21 14 PM" src="https://user-images.githubusercontent.com/105544091/172144285-7685ff7f-7663-4297-a66e-75cbcf5ee431.png">

위와 같이 './UIController' 부분이 빨간색이 뜨고요, 에러를 수정해보려해도 수정할 수 없다고 뜹니다. 

제가 현재 생각하는 우선 수정해야할 부분은 

export default class GetItem (이부분을 coin) 으로 수정 extends ZepetoScriptBehaviour 로 바꿔줘야하지 않나 정도 밖에 생각이 안되는데,,  혹시 고쳐야 할 부분 아시는 고수님들 계시면 설명 부탁드리겠습니다,,! 

...ㅠ 

++ 그리고 유투브에 찾아봐도 코인먹기 자료는 찾을 수가 없어서.. 혹시 관련 영상이나 링크 있으시면 공유 부탁드리겠습니다. 

감사합니다!

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 동일한 경로에 UIController라는 스크립트가 없을 시 에러가 뜹니다.
파일 경로 확인 부탁 드리며, 샘플 월드에서 코드만 보기 보다는
프로젝트를 전체적으로 다운받아서 실행하며 구조를 익혀보시기를 추천 드립니다💪

**Reply by almondsky** [reply]
Evangelist님 늘 감사드립니다. 한가지만 더 여쭤볼께요 ! 
다름이 아니라, 게임 구현은 완성되서 잘 작동되고, 이제 게임 종료시 점수판이 뜨도록 하기 위해 아래와 같이 세팅이 완료된 상태입니다.
<img width="168" alt="Screen Shot 2022-06-07 at 1 32 54 PM" src="https://user-images.githubusercontent.com/105544091/172296563-705dc9c0-ea1d-45be-8a6f-4528cc0db18b.png">
점수판에 프로필 사진이 보이게 하기위해 get profile texture 도 해놓은 상태입니다. set score 과 get range rank 에는 leaders id 도 넣어놓은 상태이구요. 이렇게 세팅 후 큐알코드를 받아 모바일에서 작동해보면 게임 종료 후 점수판이 뜨지를 않는데,, 더 추가해야 할 것이 있을까요? 
참고로 get profile texture 은 현재 이렇게 user id 와 sample 은 비워놓은 상태입니다. 
<img width="269" alt="Screen Shot 2022-06-07 at 1 35 22 PM" src="https://user-images.githubusercontent.com/105544091/172296801-5a632a42-4eb6-4bb7-8f55-089eea23a934.png">
 
감사합니다!

**Reply by Zepeto-TechEvangelist** [reply]
안녕하세요, 프로필 사진이 뜨지 않는다는 것이라면 코드를 아래와 같이 작성해 보시고, 샘플 이미지 부분에 UI>Image를 연결시켜 주시기를 바랍니다.

import { ZepetoScriptBehaviour } from 'ZEPETO.Script'
import {ZepetoWorldHelper} from 'ZEPETO.World'
import {Texture,Texture2D, Sprite, Rect, Vector2} from 'UnityEngine'
import {Image} from 'UnityEngine.UI'
import { WorldService } from 'ZEPETO.World';

export default class GetTextureExample extends ZepetoScriptBehaviour {
    
    public sampleImage:Image;

    Start() {
        
        ZepetoWorldHelper.GetProfileTexture(WorldService.userId,(texture:Texture)=>{

            this.sampleImage.sprite = this.GetSprite(texture);

        },(error)=>{
            console.log(error);
        });
    }

    GetSprite(texture:Texture){
        let rect:Rect = new Rect(0, 0, texture.width, texture.height);
        return Sprite.Create(texture as Texture2D, rect, new Vector2(0.5, 0.5));
    }
}

**Reply by almondsky** [reply]
제가 말을 헷갈리게 써놨습니다 죄송합니다.! 프로필 사진도 안보이는건 맞는데, 점수판 또한 보이지 않습니다. 위와 같이 필요한 리더스보드 세팅은 다 했는 상태입니다. 어딜 손봐야 월드가 종료 된 후 유저들의 랭킹 점수판이 보일까요.? ㅜ

**Reply by Zepeto-TechEvangelist** [reply]
세팅뿐 아니라 실제로 리더보드 정보를 조회하는 코드 부분을 봐야 할 수도 있고 
UI와 연동 부분도 봐야할 것 같습니다.
아래 가이드를 참고 부탁 드립니다. 
https://studio.zepeto.me/kr/guides/searching-leaderboard-details

**Reply by almondsky** [reply]
<img width="269" alt="Screen Shot 2022-06-07 at 8 09 09 PM" src="https://user-images.githubusercontent.com/105544091/172365380-558df8dd-3637-45be-a90e-dd9ff7d3a057.png">

말씀해주신거처럼 샘플 이미지 부분에 UI>Image 연결시켜놓은 상태입니다. 
<img width="1042" alt="Screen Shot 2022-06-07 at 5 49 12 PM" src="https://user-images.githubusercontent.com/105544091/172365747-214b8474-eee1-4ac7-b613-20a98d4cdb2f.png">
<img width="845" alt="Screen Shot 2022-06-07 at 5 52 13 PM" src="https://user-images.githubusercontent.com/105544091/172365759-f58a4e30-c283-4eef-a94b-b5dffb13611f.png">
<img width="830" alt="Screen Shot 2022-06-07 at 5 52 38 PM" src="https://user-images.githubusercontent.com/105544091/172365762-557a58c8-3eb7-4273-813d-2a177946339b.png">
<img width="777" alt="Screen Shot 2022-06-07 at 5 52 50 PM" src="https://user-images.githubusercontent.com/105544091/172365769-1d5edf83-b09f-4360-ac8d-d62926072353.png">

이렇게 스크립트는 해놓은 상태이구요.. 이렇게 하면 월드가 종료된 후에 이런 
<img width="854" alt="Screen Shot 2022-06-07 at 8 08 14 PM" src="https://user-images.githubusercontent.com/105544091/172365968-dabf98af-9f70-4daf-a90a-06ace60ee7c3.png">

점수판을 예상하였는데,, 월드가 종료후에는 제가 만들어놓은 다시 시작하기 버튼만 뜨지 점수판은 뜨지 않습니다.

**Reply by Zepeto-TechEvangelist** [reply]
console.log에는 점수가 뜨고 있나요? 그렇다면 기능적으로는 잘 동작하는 것입니다.
점수판을 위해서는 리더보드의 점수 데이터를 UI에 넣는 작업을 해야 합니다.
